src/Adapter/Product/Combination/Repository/CombinationMultiShopRepository.php

```diff
@@ -0,0 +1,668 @@
+/**
+ * @todo: This class has been added while we progressively migrate each domain to multishop It contains the new
+ *        dedicated function bound with multishop When everything has been migrated they will be moved back to
+ *        the initial CombinationRepository and single shop methods should be removed But since this will be done
+ *        in several PRs for now it's easier to separate them into two services
+ *        This is why a lot of code is duplicated between the two classes but don't worry this one is only temporary
+ */
+class CombinationMultiShopRepository extends AbstractMultiShopObjectModelRepository
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @var CombinationValidator
+     */
+    private $combinationValidator;
+
+    /**
+     * @var AttributeRepository
+     */
+    private $attributeRepository;
+
+    /**
+     * @var ProductMultiShopRepository
+     */
+    private $productRepository;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param CombinationValidator $combinationValidator
+     * @param AttributeRepository $attributeRepository
+     */
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix,
+        CombinationValidator $combinationValidator,
+        AttributeRepository $attributeRepository,
+        ProductMultiShopRepository $productRepository
+    ) {
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+        $this->combinationValidator = $combinationValidator;
+        $this->attributeRepository = $attributeRepository;
+        $this->productRepository = $productRepository;
+    }
+
+    /**
+     * @param CombinationId $combinationId
+     * @param ShopId $shopId
+     *
+     * @return Combination
+     *
+     * @throws CoreException
+     */
+    public function get(CombinationId $combinationId, ShopId $shopId): Combination
+    {
+        /** @var Combination $combination */
+        $combination = $this->getObjectModelForShop(
+            $combinationId->getValue(),
+            Combination::class,
+            CombinationNotFoundException::class,
+            $shopId
+        );
+
+        return $combination;
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param ShopId $shopId
+     * @param bool $isDefault
+     *
+     * @return Combination
+     *
+     * @throws CannotAddCombinationException
+     */
+    public function create(ProductId $productId, ShopId $shopId, bool $isDefault = false): Combination
+    {
+        $combination = new Combination(null, null, $shopId->getValue());
+        $combination->id_product = $productId->getValue();
+        $combination->default_on = $isDefault;
+        $combination->id_shop_list = [$shopId->getValue()];
+
+        $this->addObjectModelToShop($combination, $shopId->getValue(), CannotAddCombinationException::class);
```

Comment:
As I commented earlier, the StockAvailable should be created at this moment (unless maybe they are magically created by the ObjectModel?).

And this method should accept a list of ShopIds